Update awair SensorDeviceClass to specify icon

zachberger opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change

* Update the `awair` integration to use the appropriate `SensorDeviceClass` for VOC, PM2.5, and PM10 sensors, and remove the icon override for these sensors so the `SensorDeviceClass`'s default icon is used.  
* Remove the icon override so the CO2 sensor's `SensorDeviceClass`'s default icon is used.


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: N/A
- This PR is related to issue:  N/A
- Link to documentation pull request:  N/A

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

zachberger committed:
Update awair SensorDeviceClass to specify icon

homeassistant commented:
Hi @zachberger, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;core#75385).

Once you do that we will be able to review and accept this pull request.

Thanks!

zachberger commented:
I will take a look at the test failures and update accordingly. I struggled to get the test to run locally on my M1 mac..

zachberger committed:
Update awair tests to not care about specific icon

ahayworth commented:
> I will take a look at the test failures and update accordingly. I struggled to get the test to run locally on my M1 mac..

The test helper expects a dict of attributes to assert against, and you removed that dict in some cases. Definition here: https://github.com/home-assistant/core/blob/d645e80ccd5acb92c5ee6bce30c20bc634fc3e77/tests/components/awair/test_sensor.py#L65

Rather than just removing the dict from calls to `assert_expected_properties`, you could change it to passing an empty dict. Alternatively, you could change the method signature of `assert_expected_properties` to have a default value for the `attributes` argument.

zachberger committed:
Fix awair tests to use attributes argument

zachberger commented:
I've corrected the removal of `ATTR_UNIT_OF_MEASUREMENT` and I'm passing in an empty dict. I was also able to get the test to run locally and they now pass. Thanks for the help!

MartinHjelmare commented on homeassistant/components/awair/const.py:
```diff
@@ -86,7 +86,7 @@ class AwairSensorEntityDescription(SensorEntityDescription, AwairRequiredKeysMix
         key=API_VOC,
-        icon="mdi:cloud",
+        device_class=SensorDeviceClass.VOLATILE_ORGANIC_COMPOUNDS,
         native_unit_of_measurement=CONCENTRATION_PARTS_PER_BILLION,
```
This device class currently mandates ug/m3 as unit:
https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes

zachberger committed:
Don't use SensorDeviceClass for Awair VOC

Don't use SensorDeviceClass.VOLATILE_ORGANIC_COMPOUNDS for
awair integration since awair reports PPB and
SensorDeviceClass.VOLATILE_ORGANIC_COMPOUNDS requires ug/m3,
and there is no simple conversion.

MartinHjelmare reviewed (approved):
Thanks!
